modules/library/cql2-json/src/main/java/org/geotools/filter/text/cqljson/ExpressionToCQL2Json.java

```diff
@@ -182,31 +183,46 @@ private void toPeriod(ArrayNode output, Period period) {
         }
     }
 
-    private void toDate(ArrayNode output, Object value) {
+    private void toDate(ArrayNode output, java.sql.Date value) {
```

Comment:
CQL requires `java.sql.Date` instead of the other Date classes from the JVM, which are less specific to SQL ?

Reply:
java.util.Date is actually a full timestamp, so the convention is GeoTools is to use java.sql.Date to represent simple dates, as it's unmistakably just a date without a time.
Eventually GT should switch to use the new Java date/time framework, but nobody is hard pressed to do it.